magner/lib/views/components/page-header.vue

```diff
@@ -50,16 +60,47 @@ export default defineComponent({
   setup (props) {
     const { customT } = useTranslate();
     const route = useRoute();
+    const store = useStore();
 
     const tabs = computed(() => ({
       tabs: props.header.tabs?.map((tab, index) => ({ ...tab, index: index.toString() })).filter((tab) => !props.isNew || (props.isNew && !tab.hideIfNew)) || [],
       activeIndex: props.header.tabs?.findIndex((tab) => tab.active)?.toString() || -1,
     }));
 
+    const showFiltersBtn = computed<boolean>(() => store.state.hasFilters);
+
+    const toggleFilters = () => {
+      store.dispatch('toggleFiltersCollapsed');
+    };
+
+    const clickedOutside = (e: MouseEvent) => {
+      if (showFiltersBtn.value && !store.state.filtersCollapsed) {
+        const content = document.getElementById('id-table-page-top') || undefined;
+
+        if (
+          e.target
+          && !(e.target as HTMLElement).closest('.magner-page-header_filters-btn')
+          && !isInWhiteList(e.target, content)
+        ) {
+          toggleFilters();
+        }
+      }
+    };
```

Comment:
Suggestion: move repeatable logic (`clickedOutside` function here and in `main-layout/header.vue` component) outside

Reply:
For example, it could be done as a composable. Put `onMounted` and `onBeforeUnmount` hooks there too